netket/graph/lattice.py

```diff
@@ -104,323 +111,402 @@ def get_edges(atoms_positions, cutoff, distance_atol=cutoff_tol):
+        Basis coordinates are available from :func:`~netket.graph.Lattice.basis_coords` and
+        can be resolved into an id via :func:`~netket.graph.Lattice.id_from_basis_coords`.
     """
-
+    # Initialization
+    # ------------------------------------------------------------------------
     def __init__(
         self,
-        basis_vectors,
-        extent,
+        basis_vectors: _np.ndarray,
+        extent: _np.ndarray,
         *,
-        pbc: bool = True,
-        atoms_coord=[],
+        pbc: Union[bool, Sequence[bool]] = True,
+        site_offsets: Optional[_np.ndarray] = None,
+        atoms_coord: Optional[_np.ndarray] = None,
         distance_atol: float = 1e-5,
     ):
         """
         Constructs a new ``Lattice`` given its side length and the features of the unit cell.
 
         Args:
-            basis_vectors: The basis vectors of the unit cell.
-            extent: The number of copies of the unit cell.
+            basis_vectors: The basis vectors of the lattice. Should be an array
+                of shape `(ndim, ndim)` where each `row` is a basis vector.
+            extent: The number of copies of the unit cell; needs to be an array
+                of length `ndim`.
             pbc: If ``True`` then the constructed lattice
                 will have periodic boundary conditions, otherwise
-                open boundary conditions are imposed (default=``True``).
-            atoms_coord: The coordinates of different atoms in the unit cell (default=one atom at the origin).
-            distance_atol: A KDTree algorithm finds first neighbours of the lattice, which define the edges of
-                the graph. The algorithm needs to be specified some absolute tolerance for those distances,
-                as sometimes floating point errors might cause some edges not to be detected.
+                open boundary conditions are imposed (default=`True`).
+            site_offsets: The position offsets of sites in the unit cell (one site at the origin by default).
+            distance_atol: Distance below which spatial points are considered equal for the purpose
+                of identifying nearest neighbors.
 
         Examples:
-            Constructs a rectangular 3X4 lattice with periodic boundary conditions.
-
-            >>> import netket
-            >>> g=netket.graph.Lattice(basis_vectors=[[1,0],[0,1]],extent=[3,4])
+            Constructs a Kagome lattice with 3 × 3 unit cells:
+
+            >>> import numpy as np
+            >>> from netket.graph import Lattice
+            >>> # Hexagonal lattice basis
+            >>> sqrt3 = np.sqrt(3.0)
+            >>> basis = np.array([
+            ...     [1.0, 0.0],
+            ...     [0.5, sqrt3 / 2.0],
+            ... ])
+            >>> # Kagome unit cell
+            >>> cell = np.array([
+            ...     [0.0, 0.0],
+            ...     basis[0] / 2.0,
+            ...     basis[1] / 2.0,
```

Comment:
```suggestion
            ...     basis[0] / 2.0,
            ...     basis[1] / 2.0,
            ...     (basis[0]+basis[1])/2.0
```
Use proper Wyckoff positions so it wouldn't break with the premade symmetries present and future?